packages/conform-zod/coercion.ts

```diff
@@ -73,12 +53,61 @@ export function coerceFile(file: unknown) {
 	return file;
 }
 
+function coerceNumber(value: unknown) {
+	if (typeof value !== 'string') {
+		return value;
+	}
+
+	return value.trim() === '' ? value : Number(value);
```

Comment:
Is there a reason why `Number` and `BigInt` are not `undefined` for empty strings?

Reply:
That is handled by the string coercion, which is called before number / boolean coercion. I know this is a bit implicit and it would be simpler to not having the concept of composing coercion. But it is a bit tricky to handle optional cases where we don't know the type upfront and that's why we are coercing those value with string / file by default. Composing them makes it more consistent in terms of behavior... I think 🤔

Reply:
E.g. if you override the default string coercion logic to trim the spaces and you have an optional number schema. Your number schema will always see the string trimmed because we applied the string coercion in the optional schema.

Reply:
I see, I understand.
Since `composeCoercion` is used to first perform coercion of the `coerceString` string, an empty string is defined as `undefined`, so there should be no problem. 👍